lib/instrumentation/langchain/vectorstore.js

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2024 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+const {
+  AI: { LANGCHAIN }
+} = require('../../metrics/names')
+const { LangChainVectorSearch, LangChainVectorSearchResult } = require('../../llm-events/langchain')
+const { recordEvent, shouldSkipInstrumentation } = require('./common')
+const { DESTINATIONS } = require('../../config/attribute-filter')
+const { RecorderSpec } = require('../../shim/specs')
+const { LlmErrorMessage } = require('../../llm-events/openai')
+
+function recordVectorSearchResult(output, agent, segment, pkgVersion) {
```

Comment:
do you mind putting jsdoc here?

Reply:
I've added that. I've renamed this function to `recordVectorSearch` since it'll record both of the vector search events